packages/langium-cli/src/generator/ast-generator.ts

```diff
@@ -71,7 +71,7 @@ function generateAstReflection(grammar: Grammar, interfaces: Interface[]): Gener
     );
     reflectionNode.children.push(
         'export type ', grammar.name, 'AstReference = ',
-        crossReferenceTypes.map(e => `'${e.type}:${e.feature}'`).join(' | '),
+        crossReferenceTypes.map(e => `'${e.type}:${e.feature}'`).join(' | ') || 'never',
```

Comment:
Does that work? I think arrays are always truthy, even when empty, so the length should be checked explicitly.

Reply:
Oh I see, that's tricky...